lib/timeline/component/TimeAxis.js

```diff
@@ -205,7 +207,10 @@ TimeAxis.prototype._repaintLabels = function () {
   const totalizersToAdd = this.body.totalizer && this.body.totalizer.periods ? this.body.totalizer.periods.length : 0;
   const diffInHours = (end - start) / (1000 * 60 * 60);
   const rangeColumnCount = this.body.range.props && this.body.range.props.columnCount ? this.body.range.props.columnCount : 0;
-  const columnCount = rangeColumnCount ? rangeColumnCount : diffInHours / (gap ? gap : 1) + totalizersToAdd;
+  let columnCount = rangeColumnCount ? rangeColumnCount : diffInHours / (gap ? gap : 1) + totalizersToAdd;
+  if (gap == 0 && this.body.range.options.itemsFit) {
+    columnCount = (columnCount < this.body.range.options.itemsFit.length) ? this.body.range.options.itemsFit.length : columnCount;
```

Comment:
poderia ser mais simples, subindo a condicional . por exemplo :
let columnCount = rangeColumnCount ? rangeColumnCount : diffInHours / (gap ? gap : 1) + totalizersToAdd;
const  { itemsFit } = this.body.range.option;
  if (gap == 0 && itemsFit && columnCount < itemsFit.length) {
    columnCount = itemsFit.length;
}